### DISCUSSION TITLE
How to set timeout per request?

### DISCUSSION THREAD

**Discussion by StarpTech** [nodejs/undici#854] [General]
As titled. It would be great to do

```js
const {
  statusCode,
  headers,
  trailers,
  body
} = await request('http://localhost:3000/foo', { timeout: 2000 })
```

**Comment by szmarczak** [comment]
I think there's nothing wrong with the attached code

**Reply by StarpTech** [reply]
This parameter is not available.

**Reply by ronag** [reply]
Try `headersTimeout` and/or `bodyTimeout`. Seems like they are missing in docs under https://undici.nodejs.org/#/./docs/api/Dispatcher?id=parameter-dispatchoptions. PR welcome.
